### PR TITLE
Add CancellationToken to validation methods

### LINKED ISSUE #761: Integrate cancellation tokens in validation system
There is a setting for max errors, but sometimes the validation can just take a long time. While we can explore perf issues, it would also be good to be able to cancel it using CancellationToken so all the built in ways to manage workflows with them can be used.

Inspired by #760

### PR DESCRIPTION
**Pull request by twsouthwick**
Fixes #761

### CODE REVIEW

File: src/DocumentFormat.OpenXml/Validation/DocumentValidator.cs [hunk 1]
```diff
@@ -38,10 +39,11 @@ public DocumentValidator(SchemaValidator schemaValidator, SemanticValidator sema
         /// </summary>
         /// <param name="document">The document to be validated.</param>
         /// <param name="settings">The settings to be used during validation.</param>
+        /// <param name="token"></param>
```
Comment (lindexi) [inline src/DocumentFormat.OpenXml/Validation/DocumentValidator.cs:42]: I think we should add comments to tell the developer that the `token` do nothing in .NET 3.5

Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Validation/DocumentValidator.cs:42]: Thanks. This method is private in. Net 3.5 so it won't be visible on that platform as there is no cancelation token there. Or are you referring to devs in the open xml codebase?

Comment (lindexi) [reply src/DocumentFormat.OpenXml/Validation/DocumentValidator.cs:42]: @twsouthwick Yes, for developers.

Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Validation/DocumentValidator.cs:42]: Sure Good idea - I'll document that

File: src/DocumentFormat.OpenXml/Validation/ValidationContext.cs [hunk 2]
```diff
@@ -47,7 +49,15 @@ public ValidationContext(ValidationSettings settings, ValidationCache cache)
 
         public bool Valid => Errors.Count == 0;
 
-        public bool IsCancelled => MaxNumberOfErrors > 0 && Errors.Count >= MaxNumberOfErrors;
+        public bool IsCancelled
+        {
+            get
+            {
+                _token.ThrowIfCancellationRequested();
```
Comment (lindexi) [inline src/DocumentFormat.OpenXml/Validation/ValidationContext.cs:56]: I do not recommend throwing an exception in the get method. From the name of this property, we may not expect any exception selling.

Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Validation/ValidationContext.cs:56]: Hmm good point as to not throwing there. I'll change it to a method

**Pull request merged**